source/particles/property_pool.cc

```diff
@@ -20,7 +20,8 @@ DEAL_II_NAMESPACE_OPEN
 
 namespace Particles
 {
-  const PropertyPool::Handle PropertyPool::invalid_handle = nullptr;
+  const PropertyPool::Handle PropertyPool::invalid_handle =
+    numbers::invalid_unsigned_int;
```

Comment:
Since we use `Handle = std::size_t` I wonder whether we should use `static_cast<std::size_t>(-1);` here. Or we switch the `Handle` to `types::global_vertex_index` that either selects 32 bit or 64 bit integers as requested by the user. While it might feel a bit pedantic to hedge for the case that we have a valid index of 4 billion today that overlaps with the invalid index, future systems could reach those numbers more easily.

Reply:
Yes, maybe this?
```suggestion
  const PropertyPool::Handle PropertyPool::invalid_handle =
    static_cast<Handle>(-1);
```